### PR TITLE
[CARBONDATA-2839] Add custom compaction example

### PR DESCRIPTION
**Pull request by Xaprice**
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ x ] Any interfaces changed?
 no
 - [ x ] Any backward compatibility impacted?
 no
 - [ x ] Document update required?
no
 - [ x ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ x ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
small change

### CODE REVIEW

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/CustomCompactionExample.scala [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    // scalastyle:on
+
+    // show all segments: 0,1,2,3
+    spark.sql("SHOW SEGMENTS FOR TABLE custom_compaction_table").show()
+
+    // do custom compaction, segments specified will be merged
+    spark.sql("ALTER TABLE custom_compaction_table COMPACT 'CUSTOM' WHERE SEGMENT.ID IN (1,2)")
+    spark.sql("SHOW SEGMENTS FOR TABLE custom_compaction_table").show()
+
+    CarbonProperties.getInstance().addProperty(
```
Comment (zzcclp) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/CustomCompactionExample.scala]: why set this property here?

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/6210/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7836/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6561/

**Comment by chenliang613** [comment]
retest this please

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/CustomCompactionExample.scala [hunk 2]
```diff
@@ -0,0 +1,69 @@
+package org.apache.carbondata.examples
```
Comment (chenliang613) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/CustomCompactionExample.scala]: please add the apache license header

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/CustomCompactionExample.scala [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+import java.io.File
+
+import org.apache.spark.sql.SparkSession
+
+import org.apache.carbondata.core.constants.CarbonCommonConstants
+import org.apache.carbondata.core.util.CarbonProperties
+import org.apache.carbondata.examples.util.ExampleUtils
+
+
```
Comment (chenliang613) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/CustomCompactionExample.scala]: please add the description for explaining the example.

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/CustomCompactionExample.scala [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    spark.sql("SHOW SEGMENTS FOR TABLE custom_compaction_table").show()
+
+    // do custom compaction, segments specified will be merged
+    spark.sql("ALTER TABLE custom_compaction_table COMPACT 'CUSTOM' WHERE SEGMENT.ID IN (1,2)")
+    spark.sql("SHOW SEGMENTS FOR TABLE custom_compaction_table").show()
+
+    CarbonProperties.getInstance().addProperty(
+      CarbonCommonConstants.CARBON_DATE_FORMAT,
+      CarbonCommonConstants.CARBON_DATE_DEFAULT_FORMAT)
+
```
Comment (chenliang613) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/CustomCompactionExample.scala]: After custom compaction, please query table data once to check the data if it is correct?

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7846/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6571/

**Comment by xuchuanyin** [comment]
What does ‘CUSTOM COMPACTION’ mean here?

**Comment by zzcclp** [comment]
@xuchuanyin it means that it can compact specified segments which are assigned by users:
`ALTER TABLE custom_compaction_table COMPACT 'CUSTOM' WHERE SEGMENT.ID IN (1,2)`

**Comment by Xaprice** [comment]
@xuchuanyin, ‘CUSTOM COMPACTION’ is a new compaction type in addition to MAJOR and MINOR COMPACTION. When doing custom compaction, user can directly specify segment ids to be merged.

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/CustomCompactionExample.scala (hunk 1 above)
Comment (Xaprice) [reply examples/spark2/src/main/scala/org/apache/carbondata/examples/CustomCompactionExample.scala]: This Property is set to non-default value in the beginning of method 'exampleBody' . To ensure the completeness of this test case, the property is set back to default value, though it seems to be redundant.

**Comment by Xaprice** [comment]
@chenliang613, please take a look.

**Comment by Xaprice** [comment]
retest this please

**Comment by Xaprice** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/6242/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7880/

**Comment by chenliang613** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7884/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6608/

**Comment by chenliang613** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7890/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6614/

**Comment by chenliang613** [comment]
LGTM

**Pull request closed without merging**